src/generator.rs

```diff
@@ -45,6 +45,35 @@ impl Generator {
         self.generate_from_datetime(crate::time_utils::now())
     }
 
+    /// Generate a new Ulid. Each call is guaranteed to provide a Ulid with a larger value than the
+    /// last call. The time part of the returned [`Ulid`] can be up to one millisecond in the future,
+    /// under reasonable assumptions about processor speeds.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the previously returned value was already the highest possible ulid.
+    /// This should not happen before year 91000 of the gregorian calendar at least.
+    ///
+    /// ```rust
+    /// use ulid::Generator;
+    /// let mut gen = Generator::new();
+    ///
+    /// let ulid1 = gen.generate_overflowing();
+    /// let ulid2 = gen.generate_overflowing();
+    ///
+    /// assert!(ulid1 < ulid2);
+    /// ```
+    pub fn generate_overflowing(&mut self) -> Ulid {
+        let next = Ulid::new();
+        if next > self.previous {
```

Comment:
This behavior differs from `Generator::generate()`. In the same-millisecond case `generate` will always increment while this will first try a random value. I'd like to preserve the existing behavior because it matches the [reference implementation](https://github.com/ulid/javascript/blob/a5831206a11636c94d4657b9e1a1354c529ee4e9/lib/index.ts#L160).

Reply:
I'm going to handle this one once https://github.com/dylanhart/ulid-rs/pull/75#discussion_r1497118216 is decided, as it'll impact the things to do :)